src/ds/SLlist/createSL.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * creates a new singly linked list head node object suitable for use in state
+ * 
+ * @param {{id:string}} config - destructured initialization object
+ * @param {string} [config.id='tree'] - id to give the SLlist, should be unique
+ * @return {{id:string, next:object}} - SLlist head node plain data object
+ * @property {string} id - node identifier
+ * @property {?*} data - node data
+ * @property {object<{id:string, next:object}>} next - the next SLlist node plain data object
+ *
+ * @example <caption>How to use</caption>
+ *
+ * const tree = createSLlist({ id: 'my-tree' })
+ */
+
+const createSL = ({ id = 'SL' } = {}) => {
```

Comment:
`SL` is not a very descriptive identifier. `linked-list` would be better suited and would be consistent with the rest of the library.